src/clean_air/data/extract_metadata.py

```diff
@@ -91,24 +95,32 @@ def extract_metadata(
                 spatial_extent = SpatialExtent(bounding_polygon, bounding_polygon_crs)
             else:
                 spatial_extent = SpatialExtent(bounding_polygon)
+
         if len(cube.coords('time')) == 1:
-            temporal_extent = TemporalExtent(cube.coord('time').points)
-            total_temporal_extent_list.append(cube.coord('time').points)
+            time_list = num2pydate(times=cube.coord('time').points, units=cube.coord(
```

Comment:
As a purely aesthetic point I would prefer not to have a newline straight after an opening parenthesis, but this is not strictly necessary so still happy to approve without this change.